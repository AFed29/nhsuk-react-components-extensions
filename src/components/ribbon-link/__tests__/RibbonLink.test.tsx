import React from 'react';
import { shallow } from 'enzyme';
import RibbonLink from '..';

describe('RibbonLink', () => {
  it('matches snapshot', () => {
    const coolRibbon = shallow(<RibbonLink flavour="cool">Cool</RibbonLink>);
    const mildRibbon = shallow(<RibbonLink flavour="mild">Mild</RibbonLink>);
    const hotRibbon = shallow(<RibbonLink flavour="hot">Hot</RibbonLink>);

    expect(coolRibbon.prop('className')).toBe('nhsuk-ribbon-link nhsuk-ribbon-link--cool');
    expect(mildRibbon.prop('className')).toBe('nhsuk-ribbon-link nhsuk-ribbon-link--mild');
    expect(hotRibbon.prop('className')).toBe('nhsuk-ribbon-link nhsuk-ribbon-link--hot');

    expect(coolRibbon).toMatchSnapshot();
    expect(mildRibbon).toMatchSnapshot();
    expect(hotRibbon).toMatchSnapshot();

    coolRibbon.unmount();
    mildRibbon.unmount();
    hotRibbon.unmount();
  });

  describe('RibbonLink.Bar', () => {
    it('matches snapshot', () => {
      const bar = shallow(<RibbonLink.Bar />);
      expect(bar).toMatchSnapshot();
      bar.unmount();
    });
  });
});
