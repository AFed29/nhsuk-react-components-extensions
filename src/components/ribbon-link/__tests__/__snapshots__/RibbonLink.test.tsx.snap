// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RibbonLink RibbonLink.Bar matches snapshot 1`] = `
<div
  className="nhsuk-ribbon-link__bar"
/>
`;

exports[`RibbonLink matches snapshot 1`] = `
<button
  className="nhsuk-ribbon-link nhsuk-ribbon-link--cool"
  type="button"
>
  <ArrowRightCircle
    className="nhsuk-ribbon-link__icon nhsuk-ribbon-link__icon--cool"
  />
  <span>
    Cool
  </span>
</button>
`;

exports[`RibbonLink matches snapshot 2`] = `
<button
  className="nhsuk-ribbon-link nhsuk-ribbon-link--mild"
  type="button"
>
  <ArrowRightCircle
    className="nhsuk-ribbon-link__icon nhsuk-ribbon-link__icon--mild"
  />
  <span>
    Mild
  </span>
</button>
`;

exports[`RibbonLink matches snapshot 3`] = `
<button
  className="nhsuk-ribbon-link nhsuk-ribbon-link--hot"
  type="button"
>
  <ArrowRightCircle
    className="nhsuk-ribbon-link__icon nhsuk-ribbon-link__icon--hot"
  />
  <span>
    Hot
  </span>
</button>
`;
